addons/pkg/util/tkr_util.go

```diff
@@ -49,6 +54,15 @@ func GetTKRByNameV1Alpha3(ctx context.Context, c client.Client, tkrName string)
 
 	if err := c.Get(ctx, tkrNamespaceName, tkrV1Alpha3); err != nil {
 		if apierrors.IsNotFound(err) {
+			if cluster == nil {
```

Comment:
What about the case where err == nil *and* cluster == nil ? 
do we have to handle that? 
not from the current application, but from the point of view that GetTKRByNameAlpha3 is exported and can be called by anything, which means it could get a call with cluster=nil?  and tkrNmae points to an existing tkr.

Reply:
Also is this the right place to check for cluster==nil?  could you check it before calling GetTKRByNameV1Alpha3? then you don't have to pass in cluster.  seems forced to pass in a new parameter to the function just to check if it is nill.. 

In other words, what is the expected behavior (in general) if cluster == nil? 

seems to me there is a few unit tests missing covering those scnearios. 

also, from places where GetTKRByNameV1Alpha3 is called in this patch, cluster will never == nil. It is always set to something (cluster := &clusterapiv1beta1.Cluster{}). 

Again may be remove this check?